R/profile_wrapper.R

```diff
@@ -141,26 +152,72 @@ profile_wrapper <- function(mydir, model_settings) {
     vec <- c(low, high)
     num <- sort(vec, index.return = TRUE)$ix
 
-    profile <- r4ss::profile(
-      dir = profile_dir,
-      oldctlfile = "control.ss_new",
-      newctlfile = model_settings$newctlfile,
-      linenum = model_settings$linenum,
-      string = para,
-      profilevec = vec,
-      usepar = model_settings$usepar,
-      globalpar = model_settings$globalpar,
-      parlinenum = model_settings$parlinenum,
-      parstring = model_settings$parstring,
-      saveoutput = model_settings$saveoutput,
-      overwrite = model_settings$overwrite,
-      whichruns = model_settings$whichruns,
-      prior_check = model_settings$prior_check,
-      read_like = model_settings$read_like,
-      exe = model_settings$exe,
-      verbose = model_settings$verbose,
-      extras = model_settings$extras
-    )
+    # backup original control.ss_new file for use in second half of profile
+    file.copy(file.path(profile_dir, model_settings$oldctlfile), 
+      file.path(profile_dir, "backup_oldctlfile.ss"),
+      overwrite = model_settings$overwrite)
+    # backup original par file for use in second half of profile
+    # if usepar = TRUE
+    file.copy(file.path(profile_dir, "ss.par"), 
+      file.path(profile_dir, "backup_ss.par"),
+      overwrite = model_settings$overwrite)
+
+    # loop over down, then up
+    for (iprofile in 1:2) {
+      if (iprofile == 1) {
+        # subset of requested runs which are in the "low" vector
+        whichruns <- which(vec %in% low)
+        if (!is.null(model_settings$whichruns)) {
+          whichruns <- intersect(model_settings$whichruns, whichruns)
+        } 
+        if (model_settings$verbose) {
+          message("Running profiles down from estimated value:",
+            paste(whichruns, collapse = " "),
+            "out of ", 
+            length(vec))
+        }
+      }
+      if (iprofile == 2) {
+        # subset of requested runs which are in the "low" vector
+        whichruns <- which(vec %in% high)
+        if (!is.null(model_settings$whichruns)) {
+          whichruns <- intersect(model_settings$whichruns, whichruns)
+        } 
+        if (model_settings$verbose) {
+          message("Running profiles up from estimated value:",
+            paste(whichruns, collapse = " "),
+            "out of ", 
+            length(vec))
+        }
+        # copy backup back to use in second half of profile
+        file.copy(file.path(profile_dir, "backup_oldctlfile.ss"), 
+          file.path(profile_dir, model_settings$oldctlfile))
+        # copy backup back to use in second half of profile
+        file.copy(file.path(profile_dir, "backup_ss.par"), 
+          file.path(profile_dir, "ss.par"), 
+          overwrite = model_settings$overwrite)
+      }
```

Comment:
These lines can be consolidated to
```
        file.copy(
          file.path(profile_dir, c("backup_oldctlfile.ss", "backup_ss.par")), 
          file.path(profile_dir, c(model_settings$oldctlfile, "ss.par")),
          overwrite = TRUE
        )
```
Though, I might be wrong if the call to the overwrite parameter matters here. I do not think that it does but feel free to tell me that I am wrong 🤣.